ClinGen/clincoded
# Connects to #7 ft annotation includes

# Annotation Family Fix

I used to add all families — associated with a group or not — to the annotation for the paper. @kgliu0101 told me that only *unassociated* families should be in the annotation — associated families should only exist in their groups. This fixes that.

## Testing Procedure

Make a group, an unassociated, and an associated family, then examine the owning annotation to make sure it only has the unassociated family.

1. Create a new Gene-Disease record.
2. On the Record Curation page, add a new PMID that doesn’t exist in our database, just to help you identify it later. I suggest **123**.
2. From the Record Curation page, click the **Group +** button to create a new group with any information.
3. On the Group Submit Results page, choose **No** from the dropdown menu and click the **Return to Record Curation page** button.
2. From the Record Curation page, click the **Family +** button to make a new unassociated family with any information you’d like. Note the name you gave the family.
4. On the Family Submit Results page, choose **No** from the dropdown menu and click the **Return to Record Curation page** button.
5. On the Record Curation page, go to the section of the curation palette with the group you made, and click the **Add new Family to this Group** link to make an associated family, and make any family you want, but note the name you gave it.
4. On the Family Submit Results page, choose **No** from the dropdown menu and click the **Return to Record Curation page** button.
5. Now enter the URI ```/evidence/```.
6. Look through the **reference** column and look for the PMID you added earlier. Once you find it, click that row’s UUID to bring up the annotation’s JSON.
7. At the top level of the object, look for the annotation’s ```families``` array. It has only one family in it (a JSON viewing plugin might help here). Make sure the family has a ```label``` property that matches the name of the unassociated family you made.
8. Now look for the annotation’s ```groups``` array, and within it look for the ```familyIncluded``` array.
9. ```familyIncluded``` only has one family, and its ```label``` property must match the associated family you made.


# Curation, Edit, and View Titles

Titles for these pages have been redone for more consistency:

* Group curation page
* Group edit page
* Group view page
* Unassociated family curation page
* Unassociated family edit page
* Unassociaetd family view page
* Associated family curation page
* Associated family edit page
* Associated family view page

This part of the testing procedure assumes you ran the first part of the procedure, for the annotation family fix.

## Testing Procedure

1. From the Record Curation page, click the **Group +** button. The title of the page is **Curate Group Information**, and a second line has **Group: No entry**.
2. Enter a name for the group in the **Group name** field. As you edit the name, the **No entry** disappears and keeps up with whatever you entered.
3. Click the browser back button to go back to the Record Curation page, and find the group you made earlier in the curation palette.
4. Click that group’s **Edit** button.
5. You see a page similar to the one to make a new group, but it has the title **Edit Group Information** instead. Again, the next line keeps up with any edits you make to the group name field.
6. Click the browser back button to go back to the Record Curation page, and click the **View** button for the group you made. It has the title **View Group:** *group name*. Close this window.
7. On the Record Curation page, click the **Family +** button. The title of the page is **Curate Family Information**, and a second line has **Family: No entry**.
2. Enter a name for the family in the **Family name** field. As you edit the name, the **No entry** disappears and keeps up with whatever you entered.
3. Click the browser back button to go back to the Record Curation page, and find the unassociated family you made earlier in the curation palette.
4. Click that family’s **Edit** button.
5. You see a page similar to the one to make a new family, but it has the title **Edit Family Information** instead. Again, the next line keeps up with any edits you make to the family name field.
6. Click the browser back button, then click the **View** link for the unassociated family you made. Its title shows **View Family:** *family name*.
7. Close that window, then find the associated family you made earlier and click its **Edit** link.
8. Its title shows three lines: a large **Edit Family Information** line, a **Family:** *family name* line, and a **Group association:** *group name* line. The **Family** line keeps up with edits you make to the **Family Name** field.
9. Click the browser back button to go back to the Record Curation page, then click that same family’s **View** link. Its title shows **View Family:** *family name* and a second line, **Group assocation:** *group name*.

# Curation Palette Family Display

With associated families not existing in the annotation, I updated the curation palette to not only display families from the annotation, but also to dig into the annotation’s groups to find associated families. The fact that you can see associated families in the palette shows that this works.

Currently, all unassociated families appear first, followed by associated families. Sorting within these groups, as well as sorting of the groups themselves, should be considered random for now. @selinad will come up with a sorting method, if needed, later.
